packages/components/button/src/Button.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { ElementType, HTMLProps } from 'react';
+import { cx } from 'emotion';
+import { Flex, CommonProps } from '@contentful/f36-core';
+import { ChevronDown } from '@contentful/f36-icons';
+import { Icon, IconComponent } from '@contentful/f36-icon';
+
+import type { ButtonVariant, ButtonSize } from './types';
+import { styles } from './styles';
+import { Spinner } from '@contentful/f36-spinner';
+
+export interface ButtonProps
+  extends Omit<HTMLProps<HTMLButtonElement & HTMLAnchorElement>, 'size'>,
+    CommonProps {
+  children?: React.ReactNode;
+  /**
+   * Determines style variation of Button component
+   * @default secondary
+   */
+  variant?: ButtonVariant;
+  /**
+   * Determines size variation of Button component
+   * @default medium
+   */
+  size?: ButtonSize;
+  /**
+   * Applies active styles
+   * @default false
+   */
+  isActive?: boolean;
+  /**
+   * Disabled interaction and applies disabled styles
+   * @default false
+   */
+  disabled?: boolean;
+  /**
+   * Button html type attribute
+   */
+  type?: 'submit' | 'button' | 'reset';
+  /**
+   * Adds dropdown indicator icon
+   */
+  isDropdown?: boolean;
+  /**
+   * Expects any of the icon components
+   */
+  icon?: IconComponent;
+  /**
+   * Adds loading indicator icon and disables interactions
+   */
+  isLoading?: boolean;
+  /**
+   * Forces button to take 100% of the container
+   */
+  isFullWidth?: boolean;
+}
+
+const _Button = (props: ButtonProps, ref) => {
+  const {
+    children,
+    variant = 'secondary',
+    size = 'medium',
+    href,
+    disabled,
+    type = !href ? 'button' : undefined,
+    icon,
+    isActive,
+    isDropdown,
+    isLoading,
+    isFullWidth,
+    ...otherProps
+  } = props;
```

Comment:
Shouldn't the common props be destructured here? `className` and `testId`.

Reply:
Yeah, good catch. I left them to be part of `otherProps` but `className` needs merging, and `testId` doesn't resolve to `data-test-id` automatically as I thought 👍🏼